src/App.tsx

```diff
@@ -15,36 +17,99 @@
   'Garlic',
 ];
 
+enum SortType {
+  String = 'sortByAlphabet',
+  Number = 'sortByLength',
+  Default = '',
+}
+
 export const App: React.FC = () => {
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.Default);
+  const [reversed, setReversed] = useState<boolean>(false);
+  const sortedGoods = [...goodsFromServer].sort(
+    (good1: string, good2: string) => {
+      switch (sortOrder) {
+        case 'sortByAlphabet':
+          return good1.localeCompare(good2);
+        case 'sortByLength':
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    },
+  );
+
+  const handleSortByAlphabet = useCallback(() => {
+    setSortOrder(SortType.String);
+  }, []);
+
+  const handleSortByLength = useCallback(() => {
```

Comment:
[CODE STYLE] - Good job on naming your function 'handleSortByLength'. It's clear and descriptive.